lib/processRunner.js

```diff
@@ -75,46 +46,24 @@ module.exports = function () {
       container.type = 'node'
     }
 
-    if (envFile && !fs.existsSync(envFile)) {
-      return cb('Error: ' + envFile + ' does not exist')
-    }
-
-    if (container.specific && container.specific.environment) {
-      envArgs = generateEnvironment(container.specific.environment)
-    }
-
-    if (envFile) {
-      var envFileData = dotenv.parse(fs.readFileSync(envFile))
-      envArgs += ' ' + generateEnvironment(envFileData || {})
-    }
-
-    if (isWin) {
-      toExec = envArgs + ' ' + cmd
-    } else {
-      toExec = envArgs + 'bash -c "exec ' + cmd + '"'
+    env = Object.create(process.env)
+    if (container.environment) {
+      env = _.merge(env, container.environment)
     }
 
-
-
-    toExec = handleIpAddress(container, toExec)
-
-    console.log('running: ' + toExec)
-
-    env = Object.create(process.env)
     if (mode !== 'preview') {
       var options = {cwd: cwd, env: env, stdio: [0, 'pipe', 'pipe'], detached: false}
       if (container.type === 'node' && !isWin) {
         options.stdio[3] = 'ipc'
       }
 
       if (isWin) {
-        // allows quoted args to cmd.exe
-        options.windowsVerbatimArguments = true
 
-        // /s encodes inner quotes so the whole command itself can be quoted correctly
-        child = spawn(process.env.comspec, ['/s', '/c', '"', toExec, '"'], options)
+        // allows quoted args to cmd.exe, /s encodes inner quotes so the whole command itself can be quoted correctly
+        options.windowsVerbatimArguments = true
+        child = spawn(process.env.comspec, ['/s', '/c', '"', cmd, '"'], options)
       } else {
-        child = spawn('/bin/bash', ['-c', toExec], options)
+        child = spawn('/bin/bash', ['-c', cmd], options)
```

Comment:
From discussions, I thought we we're going to spawn node directly (or whichever process) instead of using bash/com?